src/org/omegat/core/segmentation/MapRule.java

```diff
@@ -44,56 +46,99 @@
 public class MapRule implements Serializable {
 
     private static final long serialVersionUID = -5868132953113679291L;
+    private static final ILogger LOGGER = Log.getLogger(MapRule.class);
+
+    /** Language Name */
+    private String languageCode;
 
     /** creates a new empty MapRule */
     public MapRule() {
     }
 
-    /** creates an initialized MapRule */
+    /**
+     * Create initialized MapRule object.
+     * 
+     * @param language
+     *            localized language name (from segmentation.conf), or language
+     *            code (from SRX)
+     * @param pattern
+     *            language pattern such as "EN.*" or ".*"
+     * @param rules
+     *            segmentation rules.
+     */
     public MapRule(String language, String pattern, List<Rule> rules) {
-        this.setLanguage(language);
+        String code = LanguageCodes.getLanguageCodeByPattern(pattern);
+        this.setLanguage(code != null ? code : language);
         this.setPattern(pattern);
         this.setRules(rules);
     }
 
-    /** Language Name */
-    private String languageCode;
-
-    public MapRule(Languagemap languagemap, List<Rule> rules) {
```

Comment:
I removed the ctor with `Languagemap` as argument. It reduces a dependency of classes. A retrieval of fields of `languagemap` is done in the caller.